lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -976,45 +977,10 @@ instance EncodeAddress n => ToJSON (ApiAddress n) where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance ToJSON (ApiT DerivationIndex) where
-    toJSON (ApiT (DerivationIndex ix))
-        | ix >= firstHardened = toJSON (show (ix - firstHardened) <> "H")
-        | otherwise = toJSON (show ix)
-      where
-        firstHardened = getIndex @'Hardened minBound
-
+    toJSON = toJSON . toText . getApiT
 instance FromJSON (ApiT DerivationIndex) where
-    parseJSON value = ApiT <$> (parseJSON value >>= parseAsText)
-      where
-        firstHardened = getIndex @'Hardened minBound
-
-        parseAsText :: Text -> Aeson.Parser DerivationIndex
-        parseAsText txt =
-            if "H" `T.isSuffixOf` txt then do
-                DerivationIndex ix <- castNumber (T.init txt) >>= parseAsScientific
-                pure $ DerivationIndex $ ix + firstHardened
-            else
-                castNumber txt >>= parseAsScientific
-
-        parseAsScientific :: Scientific -> Aeson.Parser DerivationIndex
-        parseAsScientific x =
-            case toBoundedInteger x of
-                Just ix | ix < firstHardened -> pure $ DerivationIndex ix
-                _ -> fail $ mconcat
-                    [ "A derivation index must be a natural number between "
-                    , show (getIndex @'Soft minBound)
-                    , " and "
-                    , show (getIndex @'Soft maxBound)
-                    , "."
-                    ]
-
-        castNumber :: Text -> Aeson.Parser Scientific
-        castNumber txt =
-            case readMay (T.unpack txt) of
-                Nothing ->
-                    fail "expected a number as string with an optional 'H' \
-                         \suffix (e.g. \"1815H\" or \"44\""
-                Just s ->
-                    pure s
+    parseJSON = parseJSON
+        >=> fmap ApiT . eitherToParser . first ShowFmt . fromText
```

Comment:
Parsing delegated to the `FromText` instance and moved near the data-type declaration.